docs(airflowctl): add authentication details (#55633)
mistercrunch/Airflow: airflow-ctl/docs/howto/index.rst

is the API URL and the environment name. The token is stored in the keyring backend and is not persisted in the
configuration file. The keyring backend is used to securely store the token and is not accessible to the user.

What is authentication for ``airflowctl``?
``````````````````````````````````````````
For ``airflowctl`` to be able to communicate with the Airflow API, it needs to authenticate itself and acquire token.
This is done using either a token or a username and password.
The token can be acquired from the Airflow API or generated using a username and password.

> .. image:: ../images/diagrams/airflowctl_api_network_diagram.svg

Reviewer: Nit: how thie image was prepared ? What if we need to modify them? 

I am quite a fan of creating such images using python code (like airflow DAGs) - with `diagram` package. We have a few of those architecture diagrams generated in our docs and we even have prek hook that will automatically re=generate the images when needed:

For example: https://github.com/apache/airflow/blob/main/docs/images/documentation_architecture.py

* https://github.com/apache/airflow/blob/main/scripts/ci/prek/generate_airflow_diagrams.py

This way anyone in the future can modify the sources for such diagrams and re-run the generator to get the image reganerated...

Maybe we could use it here too ?

Reviewer 2: +1. Having used it a bit as well, it is also fun to make :)

Author: I think those are great questions Jarek :) I was thinking the same but didn't know we had this. I used draw.io and thought about adding that as well for further editing but couldn't find a similar approach. I found SVG files with diagrams. I missed this and I was unsure how they are generated and added.

It makes sense and thanks for the direction! I will update and use that approach

Author: I have updated the diagram. Removed the old one and made the new one with the script, which I also included in the `generate_airflow_diagrams` generator :)

  :target: https://raw.githubusercontent.com/apache/airflow/main/airflow-ctl/docs/images/diagrams/airflowctl_api_network_diagram.svg
  :width: 60%
  :alt: airflowctl Auth Login Help

Parameter Details for airflowctl auth login
```````````````````````````````````````````

**--api-url**: This parameter is required. (e.g. ``http://localhost:8080``)
The default value is ``http://localhost:8080``. Full URL of the Airflow API. Without any ``/api/*`` suffixes.
If you are running the ``airflowctl`` in ``breeze`` container, it is optional.

**--api-token**: This parameter is optional.

> If you neither do not provide it nor or set the environment variable ``AIRFLOW_CLI_TOKEN``, you can skip using this parameter.

Reviewer 3: This is a bit double negative. Hard to read. Proposal to re-phrase:
```suggestion
If you are setting the token via the environment variable ``AIRFLOW_CLI_TOKEN``, you can skip using this parameter.
```

**--username**: This parameter is optional.

> If you do not provide ``--api-token`` or set the environment variable ``AIRFLOW_CLI_TOKEN``, you must provide this parameter along with ``--password``.

Reviewer 3: Also hard to read, proposal:
```suggestion
If you are not using ``--api-token`` or the environment variable ``AIRFLOW_CLI_TOKEN``, you must provide a username to authentication along with ``--password``.
```

**--password**: This parameter is optional.

> If you do not provide ``--api-token`` or set the environment variable ``AIRFLOW_CLI_TOKEN``, you must provide this parameter along with ``--username``.

Reviewer 3: I propose to phrase it simpler
```suggestion
If you provide a username via ``--username`` this is the required password to authenticate.
```
